spec/fixtures/mock_upstream.lua

```diff
@@ -123,7 +123,7 @@ end
 
 local function filter_access_by_basic_auth(expected_username,
                                            expected_password)
-   local headers = ngx.req.get_headers()
+   local headers = ngx.req.get_headers(0)
```

Comment:
How could this have an impact on existing and future tests?

Reply:
@jschmid1,

If there is any test that currently sends over 100 parameters this should hi-light that the test is wrong. But there was no existing test currently (outside this PR) that actually send over 100 args. I want our mock endpoint to always return exactly what was sent to there. This makes it more exact with parameters over 100 (as it doesn't truncate them). Also it makes the mock server immune of possible monkey patching that this PR does.